tqdm.py

```diff
@@ -0,0 +1,62 @@
+import collections
+import time
+import copy
+
+
+class Tqdm(collections.Iterator):
+    """
+    Tqdm class will receive a given iterator and will wrap and analyze it's execution with aditional metadata such as
+    running time and progress status
+    """
+
+    def __init__(self, func, length=None):
+        self.func = func
+        self.index = 0
+        self.iter_length = length
+
+    def __iter__(self):
+        if hasattr(self.func, '__len__'):
+            self.iter_length = len(self.func)
+
+        self.iter = iter(self.func)
+        self.start_time = time.time()
+        return self
+
+    def __next__(self):
+        next_val = next(self.iter)
+
+        self.index += 1
+        output = f'Running iteration number {self.index}, '
+        elapsed_seconds = time.time() - self.start_time
+        output += f'Elapsed time: {elapsed_seconds:10.4} seconds, '
+
+        iterations_per_second = self.index / elapsed_seconds
+        output += f'Iterations per second: {iterations_per_second:10.4}, '
+
+        output += self._calculate_progress()
+
+        print(f'\r{output}', flush=True, end='')
+
+        return next_val
+
+    def _calculate_progress(self):
+        if self.iter_length is None:
+            return ''
+        output = '['
+        for index in range(self.iter_length):
```

Comment:
Sorry about the review ping-pong, but 1 more thing to fix:
If you have an iterator of 100,000 fast iterations, you will print 100,000 `#` every iteration.